iwyu_output.cc

```diff
@@ -1069,6 +1069,34 @@ void ProcessFullUse(OneUse* use,
   if (use->ignore_use())   // we're already ignoring it
     return;
 
+  // We normally ignore uses for builtins, but when there is a mapping defined
+  // for the symbol, we should respect that.  So, we need to determine whether
+  // the symbol has any mappings.
+  bool is_builtin_function = false;
+  if (const clang::IdentifierInfo* iden = use->decl()->getIdentifier()) {
```

Comment:
I think it would be nice to break this code out to a helper in `iwyu_ast_util` -- the flow in `ProcessFullUse` is overwhelming enough without these details. Something like `bool IsBuiltinFunction(const clang::NamedDecl* decl, const std::string& symbol_name);`